apple/internal/macos_rules.bzl

```diff
@@ -83,30 +83,59 @@ load(
     "@build_bazel_rules_apple//apple/internal/utils:clang_rt_dylibs.bzl",
     "clang_rt_dylibs",
 )
+load(
+    "@build_bazel_rules_apple//apple/internal:cc_info_support.bzl",
+    "cc_info_support",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal:swift_support.bzl",
+    "swift_support",
+)
+load(
+    "@build_bazel_rules_swift//swift:swift.bzl",
+    "SwiftInfo",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal/aspects:swift_dynamic_framework_aspect.bzl",
+    "SwiftDynamicFrameworkInfo",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal:framework_import_support.bzl",
+    "libraries_to_link_for_dynamic_framework",
+)
 load(
     "@build_bazel_rules_apple//apple:providers.bzl",
     "AppleBinaryInfo",
     "AppleBinaryInfoplistInfo",
+    "AppleFrameworkBundleInfo",
     "MacosApplicationBundleInfo",
     "MacosBundleBundleInfo",
     "MacosExtensionBundleInfo",
+    "MacosFrameworkBundleInfo",
     "MacosKernelExtensionBundleInfo",
     "MacosQuickLookPluginBundleInfo",
     "MacosSpotlightImporterBundleInfo",
+    "MacosStaticFrameworkBundleInfo",
     "MacosXPCServiceBundleInfo",
 )
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
 
 def _macos_application_impl(ctx):
     """Implementation of macos_application."""
-    embedded_targets = ctx.attr.extensions + ctx.attr.xpc_services
-
     actions = ctx.actions
     apple_mac_toolchain_info = ctx.attr._mac_toolchain[AppleMacToolsToolchainInfo]
     apple_xplat_toolchain_info = ctx.attr._xplat_toolchain[AppleXPlatToolsToolchainInfo]
     bundle_id = ctx.attr.bundle_id
     bundle_name, bundle_extension = bundling_support.bundle_full_name_from_rule_ctx(ctx)
     executable_name = bundling_support.executable_name(ctx)
-    bundle_verification_targets = [struct(target = ext) for ext in embedded_targets]
+    bundle_verification_targets = [struct(target = ext) for ext in ctx.attr.extensions]
```

Comment:
The original `bundle_verification_targets` also contain `ctx.attr.xpc_services`. Why was that removed?